lnwallet/channel_test.go

```diff
@@ -833,6 +839,122 @@ func TestCheckHTLCNumberConstraint(t *testing.T) {
 
 }
 
+// TestCheckDustLimit checks that unsettled HTLC with dust limit not included in
+// commitment transaction as output, but sender balance is decreased (thereby all
+// unsettled dust HTLCs will go to miners fee).
+func TestCheckDustLimit(t *testing.T) {
+	createHTLC := func(data, amount btcutil.Amount) (*lnwire.HTLCAddRequest,
+		[32]byte) {
+		preimage := bytes.Repeat([]byte{byte(data)}, 32)
+		paymentHash := fastsha256.Sum256(preimage)
+
+		var returnPreimage [32]byte
+		copy(returnPreimage[:], preimage)
+
+		return &lnwire.HTLCAddRequest{
+			RedemptionHashes: [][32]byte{paymentHash},
+			Amount:           amount,
+			Expiry:           uint32(5),
+		}, returnPreimage
+	}
+
+	// Create a test channel which will be used for the duration of this
+	// unittest. The channel will be funded evenly with Alice having 5 BTC,
+	// and Bob having 5 BTC.
+	aliceChannel, bobChannel, cleanUp, err := createTestChannels(3)
+	if err != nil {
+		t.Fatalf("unable to create test channels: %v", err)
+	}
+	defer cleanUp()
+
+	aliceDustLimit := aliceChannel.channelState.OurDustLimit
+	bobDustLimit := bobChannel.channelState.OurDustLimit
+
+	if  (bobDustLimit - aliceDustLimit) > 2{
```

Comment:
Aghhh... :) Thanks! Because of this multi-hop test I had started forget to check the reason of test failure...  
> I don't think this line is needed. It seems that the value of the HTLC within the test simply needs to be above Alice's dust limit, but below Bob's dust limit.
done